spec/spec_helper.rb

```diff
@@ -114,11 +115,23 @@ class Pet < ActiveRecord::Base
   col :home_address, :type => :text
   if ENV['DB'] == 'postgresql'
     col :tsntz, :type => 'timestamp without time zone'
+  else
+    add_index :name, :unique => true
+  end
+end
+if ENV['DB'] == 'postgresql'
+  begin
+    Pet.connection.execute("ALTER TABLE pets DROP CONSTRAINT IF EXISTS unique_name")
+  rescue => e
+    puts e.inspect
   end
-  add_index :name, :unique => true
 end
+
 Pet.auto_upgrade!
 
+if ENV['DB'] == 'postgresql'
```

Comment:
this sounds crazy, but i think you need to automatically run the postgres tests at least 3 times:

1. no unique constraint - so just as before, it automatically creates a custom function
2. yes unique constraint, disable_native: nil - so native kicks in automatically
3. yes unique constraint, disable_native: true - shows that unique constraint does not interfere with custom function

you should probably do this with env vars

it makes sure that 100% of the test coverage applies to both the function and the native mode

(of course on postgres 9.4, (2) and (3) are identical)